aten/src/ATen/ZeroTensorFallback.cpp

```diff
@@ -9,7 +9,41 @@
 
 namespace at {
 
-  // TODO: add a note explaining the design decisions
+ /*
+  * ZeroTensor, use O(1) memory (tensors with a storage pointing to nullptr),
+  * are primarily used to represent undefined gradients in forward AD.
+  *
+  * ZeroTensors do not propagate nans or other extreme values. This is rooted in
```

Comment:
(Thinking about that the div example we were discussing in the other PR) We can actually turn this into something more general->

"ZeroTensor today does not perfectly handle NaNs and Infs as we don't check the actual values and assume that they are non-zero, non-inf, non-NaN etc."